pkg/gcpclient/client.go

```diff
@@ -0,0 +1,186 @@
+
+	empty, err := c.cloudResourceManagerClient.Projects.Delete(c.projectName).Do()
+	if err != nil {
+		return &cloudresourcemanager.Empty{}, fmt.Errorf("gcpclient.DeleteProject.Projects.Delete %v", err)
+	}
+	return empty, nil
+}
+
+// TODO(Raf) SetQuotas
+// TODO(Raf) Enable APIs
+
+// GetServiceAccount returns a service account if it exists
+func (c *gcpClient) GetServiceAccount(accountName string) (*iam.ServiceAccount, error) {
+	resource := fmt.Sprintf("projects/%s/serviceAccounts/%s@%s.iam.gserviceaccount.com", c.projectName, accountName, c.projectName)
+	sa, err := c.iamClient.Projects.ServiceAccounts.Get(resource).Do()
+	if err != nil {
+		return &iam.ServiceAccount{}, fmt.Errorf("gcpclient.GetServiceAccount.Projects.ServiceAccounts.Get %v", err)
+	}
+
+	return sa, nil
+}
+
+// CreateServiceAccount creates a service account with required roles.
+func (c *gcpClient) CreateServiceAccount(name, displayName string) (*iam.ServiceAccount, error) {
+
+	CreateServiceAccountRequest := &iam.CreateServiceAccountRequest{
+		AccountId: name,
+		ServiceAccount: &iam.ServiceAccount{
+			DisplayName: displayName,
+		},
+	}
+
+	serviceAccount, err := c.iamClient.Projects.ServiceAccounts.Create("projects/"+c.projectName, CreateServiceAccountRequest).Do()
+	if err != nil {
+		return &iam.ServiceAccount{}, fmt.Errorf("gcpclient.CreateServiceAccount.Projects.ServiceAccounts.Create %v", err)
+	}
+
+	return serviceAccount, nil
+}
+
+func (c *gcpClient) DeleteServiceAccount(accountEmail string) error {
+	_, err := c.iamClient.Projects.ServiceAccounts.Delete(fmt.Sprintf("projects/%s/serviceAccounts/%s", c.projectName, accountEmail)).Do()
+	if err != nil {
+		return fmt.Errorf("gcpclient.DeleteServiceAccount.Projects.ServiceAccounts.Delete: %v", err)
+	}
+
+	return nil
+}
+
+func (c *gcpClient) CreateServiceAccountKey(serviceAccountEmail string) (*iam.ServiceAccountKey, error) {
+	key, err := c.iamClient.Projects.ServiceAccounts.Keys.Create(fmt.Sprintf("projects/%s/serviceAccounts/%s", c.projectName, serviceAccountEmail), &iam.CreateServiceAccountKeyRequest{}).Do()
+	if err != nil {
+		return &iam.ServiceAccountKey{}, fmt.Errorf("gcpclient.CreateServiceAccountKey.Projects.ServiceAccounts.Keys.Create: %v", err)
+	}
+	return key, nil
+}
+
+//DeleteServiceAccountKeys deletes all keys associated with the service account
+func (c *gcpClient) DeleteServiceAccountKeys(serviceAccountEmail string) error {
+	resource := fmt.Sprintf("projects/%s/serviceAccounts/%s", c.projectName, serviceAccountEmail)
+	response, err := c.iamClient.Projects.ServiceAccounts.Keys.List(resource).Do()
+	if err != nil {
+		return fmt.Errorf("gcpclient.DeleteServiceAccountKeys.Projects.ServiceAccounts.Keys.List: %v", err)
+	}
+
+	if len(response.Keys) <= 1{
+		return nil
+	}
+
+	for _, key := range response.Keys {
+		_, err = c.iamClient.Projects.ServiceAccounts.Keys.Delete(key.Name).Do()
+	}
+
+	// ensure only one key exits
```

Comment:
TODO: we will need 2 keys for key rotation. Add to readme as roadmap item to implement automatic  key rotation